quinn-proto/src/shared.rs

```diff
@@ -257,33 +270,83 @@ pub struct ServerConfig<S>
 where
     S: crypto::Session,
 {
-    /// Transport configuration to use for incoming connections
-    pub transport: Arc<TransportConfig>,
-
-    /// TLS configuration used for incoming connections.
-    ///
-    /// Must be set to use TLS 1.3 only.
+    /// Transport configuration that is used for incoming connections.
+    transport_config: Arc<TransportConfig>,
```

Comment:
I don't really like this renaming; `config.transport` seems perfectly clear to me, whereas `config.transport_config` is unnecessarily repetitive.

Reply:
Because it was a field wrapped in `Arc` I thought it could be useful to add a getter. This to reduce redundant cloning trough out the lib.

Reply:
Okay, but in that case please also remove all the cloning that's currently there so we can assess the value of adding this getter.

Reply:
I think in context the shorter name is reasonable; it's within a `...Config` struct, it has a `...Config` type, and there's a similarly named field right below it. In any case, renaming things would be better done in a separate PR.

Reply:
Okay.